docs/cicd/github-cd-template.yml

```diff
@@ -0,0 +1,97 @@
+      # The way to build the current project depends on how you scaffold it.
+      # Different folder structures require different commands set.
+      # 'npm ci' is used here to install dependencies and it depends on package-lock.json.
+      # If you prefer to use 'npm ci', please make sure to commit package-lock.json first, or just change it to 'npm install'.
+      - name: Build the project
+        run: cd tabs && npm ci && npm run build
+
+      # Run unit test.
+      # Currently, no opinioned solution for unit test provided during scaffolding, so,
+      # set up any unit test framework you prefer (for example, mocha or jest) and update the commands accordingly in below.
+      - name: Run Unit Test
+        run: cd tabs && npm run test
+
+      # We suggest to do the provision steps by case manually or in a separated workflow, so just comment the following steps for references.
+      # After provision, you should commit .fx/env.default.json into the repository for later use.
+      # You should pick required secrets from .fx/default.userdata, and save them in the repository secrets (https://docs.github.com/en/actions/reference/encrypted-secrets) which can be refered by the step with name 'Generate default.userdata'. 
+      #- name: Provision hosting environment
+      #  uses: OfficeDev/teamsfx-cli-action@v1
+      #  with:
+      #    commands: provision
+      #    subscription: ${{env.AZURE_SUBSCRIPTION_ID}}
+
+      #- name: Commit provision configs if necessary
+      #  uses: stefanzweifel/git-auto-commit-action@v4
+      #  with:
+      #    commit_message: "chore: commit provision configs"
+      #    file_pattern: .fx/env.default.json
+      
+      #- name: Upload default.userdata as artifact
+      #  uses: actions/upload-artifact@v2
+      #  with:
+      #    name: defaultUserData
+      #    path: .fx/default.userdata
+      
+      - name: Generate default.userdata
+        env:
+          USERDATA_TENANT_ID: ${{ secrets.USERDATA_TENANT_ID }}
+          USERDATA_CLIENT_SECRET: ${{ secrets.USERDATA_CLIENT_SECRET }}
+          USERDATA_BOT_PASSWORD: ${{ secrets.USERDATA_BOT_PASSWORD }}
+        run: |
+          [ ! -z "${USERDATA_TENANT_ID}" ] && echo "solution.teamsAppTenantId=${USERDATA_TENANT_ID}" >> .fx/default.userdata
+          [ ! -z "${USERDATA_CLIENT_SECRET}" ] && echo "fx-resource-aad-app-for-teams.clientSecret=${USERDATA_CLIENT_SECRET}" >> .fx/default.userdata
+          [ ! -z "${USERDATA_BOT_PASSWORD}" ] && echo "fx-resource-bot.botPassword=${USERDATA_BOT_PASSWORD}" >> .fx/default.userdata
+
+      - name: Deploy to hosting environment
+        uses: OfficeDev/teamsfx-cli-action@v1
+        with:
+          commands: deploy
+
+      # This step is to pack the Teams App as zip file,
+      # which can be used to be uploaded onto Teams Client for installation.
+      - name: Build Teams App's package
+        uses: OfficeDev/teamsfx-cli-action@v1
+        with:
+          commands: build
```

Comment:
we need to fix this lifecycle, we should be doing build, then deploy, then publish. Building after deploy before publish breaks the lifecycle model

Reply:
i guess its ok to push this out as is and change later, i am just re-iterating my feedback from before

Reply:
I agree with this. We will drive to refactor the naming of cli's sub-commands.

Reply:
@MSFT-yiz or @eriolchan Can you track this? I'll resolve the comment below. Basically - `npm build` vs `teamsfx build` is confusing if `teamsfx build` just packages the app.